### ISSUE TITLE
Symbology geometry generator: Index missmatch

### ISSUE THREAD

**Issue by vincentBenet** [opengisch/QField#5430]
### Describe the issue
On Qgis on a layer, I am using symbology based on the ID of another layer with geometry generator line to make a line based on the ID point of the other layer. 

On Qgis this is working well, but on Qfield, the line is going somewhere else (this is OK on only few points) 

### Reproduction steps
1. Create a first layer named 'sites' of points and place few points. 
2. Create a second layer of linestring geometry with an attribute id_site and place few lines with attribute based on ID of first layer. 
3. On the second layer add on symbology a geometry generator line with following formulla: make_line($geometry, geometry(get_feature_by_id('sites', "id_site"))

### Expected behavior
1. On Qgis you have a line that goes to the corresponding id_site
2. Load the project on Qfield and you should have same lines

### Observed behavior
1. On Qfield, some lines are not going to the correct id_site, it has changed compared to QGIS. 

### Screenshots and GIFs
![Screenshot_20240707_104634](https://github.com/opengisch/QField/assets/38955854/9875915c-1188-46ea-b103-eccb9abe27d5)

### Desktop (please complete the following information)
 - OS: Windows 10
 - QGIS Version 3.22.0

### Mobile (please complete the following information)
 - Device: Honor
 - OS: Android 11
 - QField version: 33603
 - 
### Additional information
* Problem started happening recently, didn't happen in an older version of QField: No
* Problem can be reliably reproduced, doesn't happen randomly: Yes
* Problem happens with all files and projects, not only some files or projects: Yes

[If the problem happened with QFieldCloud, please add your username and project name.]
Custom Qfieldcloud server from fresh build, I can create username if needed

**Comment by m-kuhn** [comment]
Can you try with the layer name instead of id?

**Comment by vincentBenet** [comment]
> Can you try with the layer name instead of id?

No, I am using the function ``get_feature_by_id('sites', "id_site")``, but the layer name is included in it.

**Comment by nirvn** [comment]
@vincentBenet , if possible, we'd benefit from having a sample project with data that already show the discrepancy between QGIS and QField.

**Comment by vincentBenet** [comment]
@nirvn here it is:

Qfield on computer:
![image](https://github.com/opengisch/QField/assets/38955854/0bf39380-11d3-47d8-9b1d-5ba272261521)

Qfield on phone:
![WhatsApp Image 2024-07-08 à 09 21 49_f5f6bc7e](https://github.com/opengisch/QField/assets/38955854/ff65cf86-d8b0-44f2-842b-c9132434d65c)

QGIS:
![image](https://github.com/opengisch/QField/assets/38955854/0e72e14b-3c9f-46ba-9687-85737577d8ae)

Project:
[peche.zip](https://github.com/user-attachments/files/16128977/peche.zip)
